src/traces/histogram2dcontour/attributes.js

```diff
@@ -23,18 +23,20 @@ module.exports = extendFlat({
 
     histnorm: histogram2dAttrs.histnorm,
     histfunc: histogram2dAttrs.histfunc,
-    autobinx: histogram2dAttrs.autobinx,
     nbinsx: histogram2dAttrs.nbinsx,
     xbins: histogram2dAttrs.xbins,
-    autobiny: histogram2dAttrs.autobiny,
     nbinsy: histogram2dAttrs.nbinsy,
     ybins: histogram2dAttrs.ybins,
 
     autocontour: contourAttrs.autocontour,
     ncontours: contourAttrs.ncontours,
     contours: contourAttrs.contours,
     line: contourAttrs.line,
-    zhoverformat: histogram2dAttrs.zhoverformat
+    zhoverformat: histogram2dAttrs.zhoverformat,
+    _deprecated: {
```

Comment:
> In the meantime, it might be best to keep the autobin(x|y) attributes out of _deprecated and simply add a note in their description.

Just so folks looking for the old `autobin(x|y)` will see the deprecation notice? What if I just mention the fate of `autobin(x|y)` in the `start`/`end`/`size` descriptions? If I put them back in the schema they'll validate as OK (unless we do something weird like make them `enumerated` with empty `values` or make `deprecated` a property of the attribute and check for it in `validate`)

Reply:
> Might be nice to open an issue about these

https://github.com/plotly/documentation/issues/1104

Reply:
> If I put them back in the schema they'll validate as OK

Good point, But I think this is what should happen in v1. We shouldn't force people to stop using `autobin(x|y)` in v1.

> Just so folks looking for the old autobin(x|y) will see the deprecation notice? 

Yes.

Reply:
Alright, good points. I brought `autobin(x|y)` back from `_deprecated` and gave all these attributes better descriptions -> https://github.com/plotly/plotly.js/pull/3044/commits/76eee4b992eacbebff1f00ee025c7696eeb7f5d3
It's a bit hacky getting this to work with `Plotly.validate` actually, because we have to make sure they make it into `fullData` when they're present in `data`, even though they're discarded immediately. But AFAICT this does that, and is thoroughly tested.

Reply:
Thanks very much for doing this!